mautic/mautic
# Change OAuth2 Package to use twig

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [X]
| BC breaks? (use the c.x branch)        | [X]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

We are using a forked repo to get th a version of `FriendsOfSymfony/FOSOAuthServerBundle` that supports PHP templates, since we are moving away from it, the old discussion is mentioned on this [thread](https://github.com/mautic/mautic/pull/10111#issuecomment-854641323)


This PR is needed to use the twig version 

Few notices on this PR: 
1- The only version of `FOSOAuthServerBundle` that supports php8 is v2 which is hosted in the `dev-master` branch on their repo, they do not have a new released version, here is their [changelog](https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/blob/master/CHANGELOG-2.0.md)
2. I removed all the references on the fork we are using. 
3. Refactored the code to match the changes on `FOSOAuthServerBundle`
4. Renamed the event listener name and that causes phpstan to fail.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test Oauth2 capabilities
